Heart/Provider/Presentation/Controllers/ProvidersController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Heart\Provider\Presentation\Controllers;
+
+use App\Http\Controllers\Controller;
+use Heart\Provider\Application\NewAccountByProvider;
+use Heart\Provider\Domain\Enums\ProviderEnum;
+use Heart\Provider\Presentation\Requests\CreateProviderRequest;
+use Illuminate\Http\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+class ProvidersController extends Controller
+{
+    public function postProvider(
+        CreateProviderRequest $request,
+        string $provider,
+        NewAccountByProvider $action,
+    ): JsonResponse {
+        $response = $action->handle(
+            ProviderEnum::from($provider),
+            $request->input('provider_id'),
+            $request->input('username')
+        );
```

Comment:
What if the database is down? Shouldn't we handle this situations and define responses for them?